WordPress/openverse-api
# Fix `length` query for audio

## Fixes


Fixes #652 by @AetherUnbound
Fixes #653 by @AetherUnbound

## Description


Audio already had a duration field that referred to the duration in milliseconds. Based on the discussion with @dhruvkb, I converted the `duration` Enum field to  a `length` field to be able to filter the search results by 'duration categories' (short, medium, long). I also added the `medium` length to the constants so that it's available as an option.

To not make this a breaking change, I added `duration` as a possible filter query parameter that falls back to `length`. However, since  the `duration` parameter never worked before, this might not be necessary. What do you think?

## Testing Instructions

Run the API using `just start` and `just init`. After the indexes are ready, query for some sound using `length=<short|medium|long>` parameter, eg. `http://localhost:8000/v1/audio/?q=vocal&length=medium,short`. You should see the query working. 

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
